src/batHashtbl.mlv

```diff
@@ -357,10 +362,108 @@ let filter_map_inplace f h =
        | Some v' -> Cons (k, v', loop next)) in
   BatArray.modify loop hc.data
 
-(*$= filter_map_inplace & ~printer:(IO.to_string (List.print Int.print))
+(*$= filter_map_inplace & ~printer
   (let h = Enum.combine (1 -- 5, 1 -- 5) |> of_enum in \
    filter_map_inplace (fun _ x -> if x>3 then Some (x+1) else None) h ; \
-   values h |> List.of_enum |> List.sort Int.compare) [5; 6]
+   to_comparable h) [5; 6]
+*)
+
+
+let merge f a b =
+  let res = create (max (length a) (length b)) in
+  let may_add_res k v1 v2 =
+    BatOption.may (add res k) (f k v1 v2) in
+  iter (fun k v1 ->
+    match find b k with
+    | exception Not_found ->
+        may_add_res k (Some v1) None
+    | v2 ->
+        may_add_res k (Some v1) (Some v2)
```

Comment:
cannot we use `may_add_res k (Some v1) (find_opt h2)`?

Reply:
Indeed!